calico_cni/calico_cni.py

```diff
@@ -202,7 +202,16 @@ def add(self):
         endpoint = self._provision_veth(endpoint)
```

Comment:
What errors can this throw?

We need to handle anything that goes wrong here and clean up the IP assignment and Endpoint from above.